homeassistant/components/telegram_bot/__init__.py

```diff
@@ -378,7 +379,7 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     for p_config in domain_config:
         # Each platform config gets its own bot
-        bot = initialize_bot(hass, p_config)
+        bot = await hass.async_add_executor_job(initialize_bot, hass, p_config)
```

Comment:
There are async API, eg issue creation, calls inside `initialize_bot` those will need to be updated to use the sync versions of the API.

Reply:
Just for clarification: It was ok before adding the `async_add_executor_job`, because it was executed within the main event loop, right?

Reply:
Yes, exactly.